lib/res_util/es_testdata.cpp

```diff
@@ -100,6 +106,51 @@ matrix_type * safe_copy(const matrix_type * m) {
 }
 
 
+matrix_type * matrix_delete_column(const matrix_type * m1, int column) {
+  matrix_type * m2 = matrix_alloc(matrix_get_rows(m1), matrix_get_columns(m1) - 1);
+  if (column > 0)
+    matrix_copy_block(m2, 0, 0,
+                      matrix_get_rows(m2), column,
+                      m1, 0, 0);
+
+  if (column < (matrix_get_columns(m1) - 1))
+    matrix_copy_block(m2, 0, column,
+                      matrix_get_rows(m2), matrix_get_columns(m2) - column,
+                      m1, 0, column + 1);
+
+  return m2;
+}
+
+
+matrix_type * matrix_delete_row(const matrix_type * m1, int row) {
+  matrix_type * m2 = matrix_alloc(matrix_get_rows(m1) - 1, matrix_get_columns(m1));
+  if (row > 0)
+    matrix_copy_block(m2, 0, 0,
+                      row, matrix_get_columns(m2),
+                      m1, 0, 0);
+
+  if (row < (matrix_get_rows(m1) - 1))
+    matrix_copy_block(m2, row, 0,
+                      matrix_get_rows(m2) - row, matrix_get_columns(m2),
+                      m1, row + 1, 0);
+
+  return m2;
+}
+
+matrix_type * matrix_delete_row_column(const matrix_type * m1, int row_column) {
+  matrix_type * m2 = matrix_delete_column(m1, row_column);
+  matrix_type * m3 = matrix_delete_row(m2, row_column);
+  matrix_free(m2);
+  return m3;
+}
+
+
+matrix_type * swap_matrix(matrix_type * old_matrix, matrix_type * new_matrix) {
```

Comment:
This is not really a `swap`-function, but more of an overwrite function. We should find it a better name...